legend-engine-test-runner-mapping/src/main/java/org/finos/legend/engine/test/runner/mapping/extension/MappingTestRunner.java

```diff
@@ -0,0 +1,241 @@
+
+public class MappingTestRunner implements TestRunner
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(MappingTestRunner.class);
+    private Mapping pureMapping;
+    private final MutableList<PlanGeneratorExtension> extensions;
+    private final Root_meta_pure_runtime_Runtime_Impl runtime;
+    private final PlanExecutor executor;
+    private final String pureVersion;
+    private final MutableList<ConnectionFactoryExtension> factories = org.eclipse.collections.api.factory.Lists.mutable.withAll(ServiceLoader.load(ConnectionFactoryExtension.class));
+
+    public MappingTestRunner(Mapping pureMapping, String pureVersion)
+    {
+        this.pureMapping = pureMapping;
+        this.pureVersion = pureVersion;
+        this.executor = PlanExecutor.newPlanExecutorWithAvailableStoreExecutors();
+        this.extensions = Lists.mutable.withAll(ServiceLoader.load(PlanGeneratorExtension.class));
+        this.runtime = new Root_meta_pure_runtime_Runtime_Impl("");
+    }
+
+    @Override
+    public TestResult executeAtomicTest(Root_meta_pure_test_AtomicTest atomicTest, PureModel pureModel, PureModelContextData pmcd)
+    {
+        throw new UnsupportedOperationException("Mapping Test should be executed in context of Mapping Test Suite only");
+    }
+
+    @Override
+    public List<TestResult> executeTestSuite(Root_meta_pure_test_TestSuite testSuite, List<AtomicTestId> atomicTestIds, PureModel pureModel, PureModelContextData pmcd)
+    {
+        List<org.finos.legend.engine.protocol.pure.v1.model.test.result.TestResult> results = Lists.mutable.empty();
+        RichIterable<? extends Root_meta_pure_extension_Extension> routerExtensions = extensions.flatCollect(e -> e.getExtraExtensions(pureModel));
+        MutableList<PlanTransformer> planTransformers = extensions.flatCollect(PlanGeneratorExtension::getExtraPlanTransformers);
+        ConnectionVisitor<org.finos.legend.pure.m3.coreinstance.meta.pure.runtime.Connection> connectionVisitor = new ConnectionFirstPassBuilder(pureModel.getContext());
+        org.finos.legend.engine.protocol.pure.v1.model.packageableElement.mapping.Mapping mapping = ListIterate.detect(pmcd.getElementsOfType(org.finos.legend.engine.protocol.pure.v1.model.packageableElement.mapping.Mapping.class), ele -> ele.getPath().equals(getElementFullPath(this.pureMapping, pureModel.getExecutionSupport())));
+        MappingTestSuite suite = ListIterate.detect(mapping.testSuites, ts -> ts.id.equals(testSuite._id()));
+        List<String> testIds = ListIterate.collect(atomicTestIds, testId -> testId.atomicTestId);
+        List<Pair<Connection, List<Closeable>>> connections = createConnnectionAndCloseablePair(pmcd, suite.storeTestDatas);
+        connections.stream().forEach(conn -> this.runtime._connectionsAdd(conn.getOne().accept(connectionVisitor)));
+        try
+        {
+            for (Test test : suite.tests)
+            {
+                if (testIds.contains(test.id))
```

Comment:
if we don't pass in any testIds, does it execute all tests?

Reply:
no, we wont execute anything , is no id passed in the list

Reply:
I believe you updated this to run all tests if ids are empty. 
I think you had it right earlier, since we add all test ids centrally if client provides empty list. Can we please move to earlier state